### ISSUE TITLE
LFR bias mitigation method always returns a dataset with single class when passing multiclass dataset

### ISSUE THREAD

**Issue by pradeepdev-1995** [Trusted-AI/AIF360#440]
I am trying to implement the LFR pre-processing bias mitigation in a multi-class classification dataset. The original dataset has 4 distinct classes as the target variable. Used the below code for LFR.
```
TR = LFR(unprivileged_groups=unprivileged_groups,
         privileged_groups=privileged_groups,
         k=10, Ax=0.1, Ay=1.0, Az=2.0,
         verbose=1
        )
TR = TR.fit(original_dataset, maxiter=5000, maxfun=5000)
dataset_transf_train = TR.transform(original_dataset)

print("dataset_transf_train")
print(dataset_transf_train.labels)
```
This code shows that the transformed dataset has always a single target column. Where have the other classes gone? Should I apply any tuning method for this?

**Comment by rajnairdevs** [comment]
Faced similar issue but when the threshold value is tuned, for a certain range LFR did work and brought the bias down.

# Learning Fair Representations
LFR_model = LFR(seed=1,unprivileged_groups=unprivileged_groups, privileged_groups=privileged_groups, k=3, verbose=0)
dataset_transformed = LFR_model.fit_transform(dataset_original, maxiter=5000, maxfun=5000, **threshold=0.25**)

**Comment by Rajesh-Nair** [comment]
Faced similar issue but when the threshold value is tuned, for a certain range LFR did work and brought the bias down.

Learning Fair Representations
LFR_model = LFR(seed=1,unprivileged_groups=unprivileged_groups, privileged_groups=privileged_groups, k=3, verbose=0)
dataset_transformed = LFR_model.fit_transform(dataset_original, maxiter=5000, maxfun=5000, threshold=0.25)
